src/read/pdb/utils.rs

```diff
@@ -0,0 +1,40 @@
+pub(crate) fn fast_parse_u64_from_string(s: &str) -> u64 {
+    let bytes = s.as_bytes();
+    let mut result: u64 = 0;
+    let mut i = 0;
+    let len = bytes.len();
+
+    // Skip leading whitespace
+    while i < len && bytes[i].is_ascii_whitespace() {
+        i += 1;
+    }
+
+    // Parse digits
+    while i < len && bytes[i].is_ascii_digit() {
+        result = result * 10 + (bytes[i] - b'0') as u64;
+        i += 1;
+    }
+
+    result
+}
+
+//TODO: Implement miri tests and fuzzing
+pub(crate) fn fast_trim(s: &str) -> &str {
```

Comment:
One issue that might pop up is that you slice halfway into a character that has the byte pattern of a whitespace character either leading or trailing (yes these character exists). So checking only full characters could be a better fit. So either using `chars` or `char_indices` could potentially be better, but there is also the function `is_char_boundary` that might be of help.